packages/gatsby/cache-dir/loader.js

```diff
@@ -348,12 +348,20 @@ export class BaseLoader {
     return inFlightPromise
   }
 
-  // returns undefined if loading page ran into errors
+  // returns undefined if the page does not exists in cache
   loadPageSync(rawPath) {
     const pagePath = findPath(rawPath)
     if (this.pageDb.has(pagePath)) {
-      const pageData = this.pageDb.get(pagePath).payload
-      return pageData
+      const pageData = this.pageDb.get(pagePath)
+
+      if (pageData.payload) {
+        return pageData.payload
+      }
+
+      return {
+        error: pageData.error,
+        status: pageData.status,
+      }
```

Comment:
This is a behavior change.

Reply:
I think change itself make sense but we have couple of places where we call this other than `ensure-resources` - including exposing it in public API ( https://github.com/gatsbyjs/gatsby/blob/ec57fb2e8b92c99db991e226383cfd7642eb511f/packages/gatsby/cache-dir/api-runner-browser.js#L27-L29 ), but also other places in our runtime - would it make sense to make it optional to behave as propsoed here - i.e. `loadPageSync(rawPath, { getErrorDetailsInsteadOfUndefined: true })`?

Reply:
I was thinking of that too but wasn't sure if it's worth it :shrug: